extensions/interactions/InteractiveMap/directives/InteractiveMap.js

```diff
@@ -62,10 +82,38 @@ oppia.directive('oppiaInteractiveInteractiveMap', [
           $scope.mapOptions = {
             center: new google.maps.LatLng(coords[0], coords[1]),
             zoom: zoomLevel,
-            mapTypeId: google.maps.MapTypeId.ROADMAP
+            mapTypeId: google.maps.MapTypeId.ROADMAP,
+            draggable: $scope.interactionIsActive
+          };
+
+          $scope.setOverlay = function() {
+            if ($scope.interactionIsActive) {
+              return;
+            }
+            $scope.overlayStyle = {
+              'background-color' : 'black'
+            };
+            $scope.mapStyle = {
+              opacity : '0.8'
+            };
+          };
+
+          $scope.resetOverlay = function() {
```

Comment:
It's not clear to me what set/reset mean. Should this be hideOverlay/removeOverlay instead?